packages/plugin-ext/src/plugin/notification.ts

```diff
@@ -32,13 +33,20 @@ export class NotificationExtImpl implements NotificationExt {
         options: ProgressOptions,
         task: (progress: Progress<{ message?: string; increment?: number }>, token: CancellationToken) => PromiseLike<R>
     ): Promise<R> {
+        const source = new CancellationTokenSource();
         const id = new Deferred<string>();
```

Comment:
The handling of this ID - from the old code - is very odd to me. I don't think the Ext side should let the Main side tell it what local handle to use; it should tell the Main side what handle to use to communicate with it. Similarly, the Main side waits for the `ProgressService` to provide it with a handle, when it should use its own handle and map it to what the progress service returns. But those existing oddities don't need to be fixed here.

Reply:
Yes I agree. I noticed this while making my changes as I was comparing with the VS Code implementation. It seemed that the changes required to align with VS Code would be more complex and should be a separate PR, so I chose to work within the current interface.